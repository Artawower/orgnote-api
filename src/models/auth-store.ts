import type { Ref } from 'vue';
import { Store } from './store';
import { PersonalInfo } from './user';
import { OAuthProvider } from './oauth-provider';

export interface AuthStore {
  token: Ref<string>;
  user: Ref<PersonalInfo>;
  provider: Ref<OAuthProvider>;
  auth: (params: {
    provider: string;
    environment?: string;
    redirectUrl?: string;
  }) => Promise<void>;

  authViaGithub: (redirectUrl: string) => Promise<void>;
  logout: () => Promise<void>;
  verifyUser: () => Promise<void>;
  authUser: (u: PersonalInfo, token: string) => Promise<void>;
  subscribe: (token: string, email?: string) => Promise<void>;
  removeUserAccount: () => Promise<void>;
}

export type AuthStoreDefinition = Store<AuthStore>;
