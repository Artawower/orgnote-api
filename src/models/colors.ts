// TODO: feat/stable-beta generate
export type Colors = '';
