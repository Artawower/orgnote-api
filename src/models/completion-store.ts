import { ComputedRef } from 'vue';
import { Completion, CompletionConfig } from './completion';
import { Store } from './store';

export interface CompletionStore {
  restore: () => void;
  close: () => void;
  closeAll: () => void;
  open: <T>(config: CompletionConfig<T>) => Promise<void>;
  activeCompletion?: ComputedRef<Completion>;
  nextCandidate: () => void;
  previousCandidate: () => void;
  search: (limit?: number, offset?: number) => void;
}

export type CompletionStoreDefinition = Store<CompletionStore>;
