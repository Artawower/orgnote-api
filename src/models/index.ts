export * from './note';
export * from './command';
export * from './completion';
export * from './extension';
export * from './theme-variables';
export * from './widget';
export * from './modal';
export * from './widget-type';
export * from './editor';
export * from './default-commands';
export * from './encryption';
export * from './file-system';
export * from './sync';
export * from './file-info';
export * from './file-path';
export * from './file-tree';
export * from './user';
export * from './oauth-provider';
export * from './orgnote-config';
export * from './platform-specific';
export * from './css-utils';
export * from './i18n-keys';
export * from './vue-component';
export * from './file-upload';
export * from './confirmation-modal';
export * from './pane';

// Composables
export * from './splash-screen';

// Stores
export * from './sync-store';
export * from './files-store';
export * from './file-opener-store';
export * from './file-manager-store';
export * from './auth-store';
export * from './commands-store';
export * from './commands-group-store';
export * from './extension-store';
export * from './file-system-store';
export * from './settings-store';
export * from './encryption-store';
export * from './ui-store';
export * from './sidebar-store';
export * from './toolbar-store';
export * from './modal-store';
export * from './settings-ui-store';
export * from './completion-store';
export * from './pane-store';
export * from './file-system-manager-store';

// Repositories
export * from './repositories';
