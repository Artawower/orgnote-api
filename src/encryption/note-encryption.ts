import {
  ModelsPublicNote,
  ModelsPublicNoteEncryptionTypeEnum,
} from '../remote-api';

import {
  decryptViaKeys,
  decryptViaPassword,
  encryptViaKeys,
  encryptViaPassword,
} from './encryption';
import { OrgNoteEncryption } from '../models/encryption';
import { parse, withMetaInfo } from 'org-mode-ast';
import { isGpgEncrypted } from '..';

export type NoteEncryptionType = `${ModelsPublicNote['encryptionType']}`;

export interface AbstractEncryptedNote {
  encryptionType?: NoteEncryptionType;
  encrypted?: boolean;
  meta: {
    [key: string]: any;
    published?: boolean;
  };
}

export type EncryptionResult<T> = Promise<[T, string]>;

// TODO: master change signature for encrypt notes without content
export async function encryptNote<T extends AbstractEncryptedNote>(
  note: T,
  noteText: string,
  encryptionParams: OrgNoteEncryption
): EncryptionResult<T> {
  note.encrypted = false;
  if (
    !encryptionParams.type ||
    encryptionParams.type === ModelsPublicNoteEncryptionTypeEnum.Disabled ||
    note.meta.published
  ) {
    return [note, noteText];
  }

  note.meta = { id: note.meta.id, published: note.meta.published };

  const [encryptedNote, encryptedNoteText] =
    encryptionParams.type === ModelsPublicNoteEncryptionTypeEnum.GpgKeys
      ? await encryptNoteViaKeys(
          note,
          encryptionParams.publicKey,
          encryptionParams.privateKey,
          encryptionParams.privateKeyPassphrase
        )
      : await encryptNoteViaPassword(note, noteText, encryptionParams.password);

  encryptedNote.encrypted = true;
  return [encryptedNote, encryptedNoteText];
}
export async function encryptNoteViaPassword<T extends AbstractEncryptedNote>(
  note: T,
  noteText: string,
  password: string
): Promise<[T, string]> {
  const encryptedNoteText = await encryptViaPassword(noteText, password);
  return [
    {
      ...note,
      encrypted: ModelsPublicNoteEncryptionTypeEnum.GpgPassword,
    },
    encryptedNoteText,
  ];
}

export async function encryptNoteViaKeys<T extends AbstractEncryptedNote>(
  note: T,
  noteText: string,
  publicKey: string,
  privateKey: string,
  privateKeyPassphrase?: string
): Promise<[T, string]> {
  const encryptedNoteText = await encryptViaKeys(
    noteText,
    publicKey,
    privateKey,
    privateKeyPassphrase
  );

  return [
    {
      ...note,
      encrypted: ModelsPublicNoteEncryptionTypeEnum.GpgKeys,
    },
    encryptedNoteText,
  ];
}

export async function decryptNote<T extends AbstractEncryptedNote>(
  note: T,
  noteText: string,
  encryptionParams: OrgNoteEncryption
): EncryptionResult<T> {
  const isContentEncrypted = isGpgEncrypted(noteText);
  if (
    note.meta.published ||
    !note.encryptionType ||
    !encryptionParams.type ||
    encryptionParams.type === ModelsPublicNoteEncryptionTypeEnum.Disabled ||
    !isContentEncrypted
  ) {
    note.encrypted = isContentEncrypted;
    return [note, noteText];
  }
  note.encrypted = true;
  const [decryptedNote, decryptedNoteText] =
    encryptionParams.type === ModelsPublicNoteEncryptionTypeEnum.GpgKeys
      ? await decryptNoteViaKeys(
          note,
          noteText,
          encryptionParams.privateKey,
          encryptionParams.privateKeyPassphrase
        )
      : await decryptNoteViaPassword(note, noteText, encryptionParams.password);

  const parsed = withMetaInfo(parse(decryptedNoteText));

  return [
    {
      ...decryptedNote,
      encrypted: false,
      meta: parsed.meta,
    },
    decryptedNoteText,
  ];
}

export async function decryptNoteViaPassword<T extends AbstractEncryptedNote>(
  note: T,
  noteText: string,
  password: string
): EncryptionResult<T> {
  const decryptedNoteText = await decryptViaPassword(noteText, password);
  return [
    {
      ...note,
      encrypted: ModelsPublicNoteEncryptionTypeEnum.GpgPassword,
    },
    decryptedNoteText,
  ];
}

export async function decryptNoteViaKeys<T extends AbstractEncryptedNote>(
  note: T,
  noteText: string,
  privateKey: string,
  privateKeyPassphrase?: string
): EncryptionResult<T> {
  const decryptedNoteText = await decryptViaKeys(
    noteText,
    privateKey,
    privateKeyPassphrase
  );
  return [
    {
      ...note,
      encrypted: ModelsPublicNoteEncryptionTypeEnum.GpgKeys,
    },
    decryptedNoteText,
  ];
}
