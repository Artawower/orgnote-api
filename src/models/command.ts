import { COMMAND_GROUPS } from 'src/constants';
import { DefaultCommands } from './default-commands';
import { OrgNoteApi } from 'src/api';

export type CommandGroup =
  | (typeof COMMAND_GROUPS)[number]
  | (string & Record<never, never>);

export type CommandName = DefaultCommands | (string & {});

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export interface CommandHandlerParams<T = any> {
  event?: KeyboardEvent;
  data?: T;
  meta: CommandMeta<T>;
  [key: string]: unknown;
}

export interface CommandPreview {
  description?: string;
  command?: CommandName;
  title?: string | (() => string);
  icon?: string | (() => string);
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export interface CommandMeta<T = any> extends Partial<CommandPreview> {
  /* Where is this command available, default value is global */
  group?: CommandGroup;
  allowOnInput?: boolean;
  interactive?: boolean; // TODO: add support for interactive commands
  /* When command is system command, it will not be shown for users */
  system?: boolean;
  disabled?: () => boolean;
  isActive?: () => boolean;
  context?: {
    [key: string]: T;
  };
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export interface Command<T = any, R = unknown> extends CommandMeta<T> {
  /* arguments depend on the current scope */
  handler: (
    api: OrgNoteApi,
    params?: CommandHandlerParams<T>
  ) => R | Promise<R>;
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type CommandCallback = <T = any>(meta: Command, data: T) => void;
