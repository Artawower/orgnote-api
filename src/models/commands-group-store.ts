import { Ref } from 'vue';
import { Command, CommandGroup } from './command';
import { Store } from './store';

export interface CommandsGroupStore {
  activateGroup: (group: CommandGroup) => void;
  deactivateGroup: (group: CommandGroup) => void;
  getCommandsByGroup: (group: CommandGroup) => Command[];
  currentGroupsCommands: Ref<Command[]>;
  currentGroups: Ref<CommandGroup[]>;
}

export type CommandsGroupStoreDefinition = Store<CommandsGroupStore>;
