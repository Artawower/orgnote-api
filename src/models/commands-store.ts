import { Ref } from 'vue';
import { Command, CommandCallback } from './command';
import { Store } from './store';

export type CommandUnsubscribFn = () => void;

export interface CommandsStore {
  add: (...newCommands: Command[]) => void;
  remove: (...commandsToUnregister: Command[]) => void;
  get: (name: string) => Command | undefined;
  commands: Ref<Command[]>;
  execute: (name: string, data?: unknown) => Promise<void>;
  afterExecute: (
    commandNames: string | string[],
    callback: CommandCallback
  ) => CommandUnsubscribFn;
}

export type CommandsStoreDefinition = Store<CommandsStore>;
