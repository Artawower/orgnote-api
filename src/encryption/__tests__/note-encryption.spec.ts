import { expect, test } from 'vitest';
import { decryptNote, encryptNote } from '../note-encryption';
import { Note } from '../../models';
import {
  armoredPublicKey,
  armoredPrivateKey,
  privateKeyPassphrase,
} from './encryption-keys';

test('Should encrypt note content via password', async () => {
  const content = `#+ID: qweqwe

#+TITLE: Hello worlld

* Hello?`;

  const note: Note = {
    id: 'id',
    encrypted: 'gpgPassword',
    meta: {
      title: 'My note title',
      images: [],
      published: false,
      description: 'Awesome description',
    },
    content,
    author: {
      id: '1',
      name: 'John Doe',
      email: 'test@mail.com',
    },
  };

  const encryptedNote = await encryptNote(note, {
    type: 'gpgPassword',
    password: '123',
  });

  expect(encryptedNote.content.startsWith('-----BEGIN PGP MESSAGE-----')).toBe(
    true
  );
  delete encryptedNote.content;
  expect(encryptedNote).toMatchSnapshot();
});

test('Should decrypt note content via password', async () => {
  const note: Note = {
    id: 'id',
    encrypted: 'gpgPassword',
    meta: {
      title: 'My note title',
      images: [],
      published: false,
      description: 'Awesome description',
    },
    content: `-----BEGIN PGP MESSAGE-----

wy4ECQMI/CCaKMJEqy/gyROJeRgW9I738dDFBltFlxIjhxrN7nQ6gkX4GgX6
zt3v0mABvaBJA7corlU8su21TpPApOs/+DMWpVlbI3Zer7QfQK1fSBoSTbCR
2k3g68Afayke2nLkDNkH62tdOPiTkx7bSlp1zL4uU440IM1g6dC72JkmtoTJ
5Bjlwt4ZhxFsh+c=
=csID
-----END PGP MESSAGE-----`,
  };

  const decryptedNote = await decryptNote(note, {
    type: 'gpgPassword',
    password: '123',
  });

  expect(decryptedNote).toMatchSnapshot();
});

test('Should encrypt note via keys', async () => {
  const content = `#+ID: qweqwe
#+TITLE: Hello worlld

* Hello?`;

  const note: Note = {
    id: 'id',
    encrypted: 'gpgKeys',
    meta: {
      title: 'My note title for encryption via keys',
      images: [],
      published: false,
      description: 'Awesome description',
    },
    content,
    author: {
      id: '1',
      name: 'John Doe',
      email: 'test@mail.com',
    },
  };

  const encryptedNote = await encryptNote(note, {
    type: 'gpgKeys',
    publicKey: armoredPublicKey,
    privateKey: armoredPrivateKey,
    privateKeyPassphrase,
  });

  expect(encryptedNote.content.startsWith('-----BEGIN PGP MESSAGE-----')).toBe(
    true
  );
  delete encryptedNote.content;
  expect(encryptedNote).toMatchSnapshot();
});

test('Should decrypt note via provided keys', async () => {
  const note: Note = {
    id: 'id',
    encrypted: 'gpgKeys',
    meta: {
      title: 'My note title for decryption via keys',
      images: [],
      published: false,
      description: 'Awesome description',
    },
    content: `-----BEGIN PGP MESSAGE-----

wcFMA/vryg+TTn0rAQ//TBFRjKmjRQoLhSrgeH+NbsZXbxvo7Ey4k+BQ9XA5
+CMpXH9uFUxsSaI5+McUSEt32VI17HRpXQDCL9nwaWqWOanMaRe0tXXhtox2
gfe2f/6zsge9ux+mXF2BG4z+V5T8XIOrfzxosVprdJHZHM3x7cW5USQ0t2i2
FiOUWxSZO1899J3yICpMvhDXvTLVZuKpSNQho5PyXSeZa83eN+uYkhgt9lsk
0KW88Nr435S6n6mVw/zpitXIgEKpkqh46mhG+1W3aC0lYx6j9lHm3bjtdb4r
2mtZbWKwfdBggEy8qEeiUeslvKd6uWtEccomzFgJkaqWVGknLmrBdHfztRIV
fbZhxHz+J3GFOIgXf/5+fv+zg0nTazgz1mDUfnTHw3+qcAyEJ0ADdyg9EZc+
bKbK0GwzMDPQeM+cCRDWUTiIa1ruyEETiwMdhWUDztF8XxY2o8jXPerZ0NhG
R8l+SvAdYQQXPfxEh9lA3thkyz/Vp72pW46lUeJHGSq/fS6KIdLHLP9Z2e1J
aCFpNMgyAN+BaXwnbLZfz6k5hV8awbRScSWQLEg69D9b287SFDPOYW7OZikn
CKXL7xyQ5LrWJZN9Z/UPGjy+PdEw1SBhyluW6DQ+Sz1j4K8USTLqY89EInDl
G7AxHGujR9UkUGyUvvc71XW7jEOogiDgn1ZMoj9Y58LSweUB8vfkY3VF84Fy
zGVvLFWnFbtnWuoQC4OOXv1F6ETdmEsSMEMWKzRLDGPyqNX7FhG3cejlEuiK
EsT5oQhz00RECmnR8mJkGmJhzWmmeoQvaucFqhTOZjvYl9ivuEMBZ2jtkdHn
R6UNcNZRpCbfnl7YoikqbGBgqDFYydFGHXKHDpYhcQYWJsMDUayzUiPtzmaE
tgfTgNNqgPxkLnWA99KYNU8DH+FwgaYBuw9dwdqwcjxSbZjHhCFGUfsqM+ik
O9gcoAwVO1usODlESU7OWuSF2tIv5DBG5rhlSyxBp4d4aWmaumTswAdojy2O
sM70ETbg0mC8niW9lNsgJp55oFmlksvzjUIit5rBEr55rtPcJSgakPpR2yvd
Q4XjybUmV0IdONMhk/OlqaBPtnA0RG+qTcm7oP1qH/m5zRA0ZplVQ5ylkQwr
LwGq5JQpJkgxxgLIrUbbtzYBShXSr5c1XXR0LIRiNgtb6s1s4mt+fbyExdJF
+ceuu+/xYrW/YTuEJpHxLiZ+aNPW5g5Y7Hbqu3hp3UL/kD44cc8JjZh18spX
p/ncojDhF2r2vhR+CndYcMkpGMV+t1pGKC8wlcFc7lb9GJASnqMGvhQmvIaG
gO/x7UjuTZSBW+kITHHZJDqYryKUv1j0CkHxIn9tWsYOpa1giOFtXX5v0AAM
AJR/s/beLlqwCsUdYnw1TkP/0u0ZK3RPio1nJ7S6ckPfsM7DqCWD8ILD8Cdr
cuzQrWaE30t5PXx53xBPO+6t5wKfDL35WHWG1Irmvz9UuT7tDS3IzwtF4ijF
PX6alTbxGnoHgZ4bG4J1wfpTNPppP1gJeVg67VqOypzdZi+SjofMWnFgRFmD
yEN8xpFUs7A9xryVZOosp9Sfe3IbBkO99sAQ7jV4EoMYk3/GKA==
=LjkG
-----END PGP MESSAGE-----`,
  };

  const decryptedNote = await decryptNote(note, {
    type: 'gpgKeys',
    publicKey: armoredPublicKey,
    privateKey: armoredPrivateKey,
    privateKeyPassphrase,
  });

  expect(decryptedNote).toMatchSnapshot();
});

test('Should not encrypt public note', async () => {
  const content = `#+ID: qweqwe
#+TITLE: Hello worlld

* Hello?`;

  const note: Note = {
    id: 'id',
    encrypted: 'gpgPassword',
    meta: {
      title: 'My note title',
      images: [],
      published: true,
      description: 'Awesome description',
    },
    content,
    author: {
      id: '1',
      name: 'John Doe',
      email: 'test@mail.com',
    },
  };

  const encryptedNote = await encryptNote(note, {
    type: 'gpgPassword',
    password: '123',
  });

  expect(encryptedNote.content.startsWith('-----BEGIN PGP MESSAGE-----')).toBe(
    false
  );
  expect(encryptedNote).toMatchSnapshot();
});

test('Should encrypt note with empty encrypted property', async () => {
  const content = `#+ID: qweqwe
#+TITLE: Hello worlld

* Hello?`;

  const note: Note = {
    id: 'id',
    meta: {
      title: 'My note title',
      images: [],
      published: false,
      description: 'Awesome description',
    },
    content,
    author: {
      id: '1',
      name: 'John Doe',
      email: 'test@mail.com',
    },
  };

  const encryptedNote = await encryptNote(note, {
    type: 'gpgPassword',
    password: '123',
  });

  expect(encryptedNote.content.startsWith('-----BEGIN PGP MESSAGE-----')).toBe(
    true
  );

  delete encryptedNote.content;
  expect(encryptedNote).toMatchSnapshot();
});
