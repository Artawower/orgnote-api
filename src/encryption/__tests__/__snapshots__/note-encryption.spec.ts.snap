// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should decrypt note content via password 1`] = `
[
  {
    "encrypted": false,
    "encryptionType": "gpgPassword",
    "id": "id",
    "meta": {
      "headings": [
        {
          "level": 1,
          "title": "Hello?",
        },
      ],
      "id": "qweqwe",
      "title": "Hello worlld",
    },
  },
  "#+ID: qweqwe
#+TITLE: Hello worlld

* Hello?",
]
`;

exports[`Should decrypt note via provided keys 1`] = `
[
  {
    "encrypted": false,
    "encryptionType": "gpgKeys",
    "id": "id",
    "meta": {
      "headings": [
        {
          "level": 1,
          "title": "Hello?",
        },
      ],
      "id": "qweqwe",
      "title": "Hello worlld",
    },
  },
  "#+ID: qweqwe
#+TITLE: Hello worlld

* Hello?",
]
`;

exports[`Should encrypt note content via password 1`] = `
{
  "author": {
    "email": "test@mail.com",
    "id": "1",
    "name": "John Doe",
  },
  "encrypted": true,
  "encryptionType": "gpgPassword",
  "id": "id",
  "meta": {
    "id": undefined,
    "published": false,
  },
}
`;

exports[`Should encrypt note via keys 1`] = `
{
  "author": {
    "email": "test@mail.com",
    "id": "1",
    "name": "John Doe",
  },
  "encrypted": true,
  "encryptionType": "gpgKeys",
  "id": "id",
  "meta": {
    "id": undefined,
    "published": false,
  },
}
`;

exports[`Should encrypt note with empty encrypted property 1`] = `
{
  "author": {
    "email": "test@mail.com",
    "id": "1",
    "name": "John Doe",
  },
  "encrypted": true,
  "id": "id",
  "meta": {
    "id": undefined,
    "published": false,
  },
}
`;

exports[`Should not decrypt note without provided encrypted type 1`] = `
[
  {
    "author": {
      "email": "test@mail.com",
      "id": "1",
      "name": "John Doe",
    },
    "encrypted": false,
    "id": "id",
    "meta": {
      "description": "Awesome description",
      "images": [],
      "published": false,
      "title": "My note title",
    },
  },
  "#+ID: qweqwe
#+TITLE: Hello worlld

* Hello?",
]
`;

exports[`Should not encrypt public note 1`] = `
{
  "author": {
    "email": "test@mail.com",
    "id": "1",
    "name": "John Doe",
  },
  "encrypted": false,
  "encryptionType": "gpgPassword",
  "id": "id",
  "meta": {
    "description": "Awesome description",
    "images": [],
    "published": true,
    "title": "My note title",
  },
}
`;

exports[`Should set not encrypted status when params type does not provided 1`] = `
[
  {
    "encrypted": true,
    "encryptionType": "gpgKeys",
    "id": "id",
    "meta": {
      "description": "Awesome description",
      "images": [],
      "published": false,
      "title": "My note title for decryption via keys",
    },
  },
  "-----BEGIN PGP MESSAGE-----

wcFMA/vryg+TTn0rAQ//TBFRjKmjRQoLhSrgeH+NbsZXbxvo7Ey4k+BQ9XA5
+CMpXH9uFUxsSaI5+McUSEt32VI17HRpXQDCL9nwaWqWOanMaRe0tXXhtox2
gfe2f/6zsge9ux+mXF2BG4z+V5T8XIOrfzxosVprdJHZHM3x7cW5USQ0t2i2
FiOUWxSZO1899J3yICpMvhDXvTLVZuKpSNQho5PyXSeZa83eN+uYkhgt9lsk
0KW88Nr435S6n6mVw/zpitXIgEKpkqh46mhG+1W3aC0lYx6j9lHm3bjtdb4r
2mtZbWKwfdBggEy8qEeiUeslvKd6uWtEccomzFgJkaqWVGknLmrBdHfztRIV
fbZhxHz+J3GFOIgXf/5+fv+zg0nTazgz1mDUfnTHw3+qcAyEJ0ADdyg9EZc+
bKbK0GwzMDPQeM+cCRDWUTiIa1ruyEETiwMdhWUDztF8XxY2o8jXPerZ0NhG
R8l+SvAdYQQXPfxEh9lA3thkyz/Vp72pW46lUeJHGSq/fS6KIdLHLP9Z2e1J
aCFpNMgyAN+BaXwnbLZfz6k5hV8awbRScSWQLEg69D9b287SFDPOYW7OZikn
CKXL7xyQ5LrWJZN9Z/UPGjy+PdEw1SBhyluW6DQ+Sz1j4K8USTLqY89EInDl
G7AxHGujR9UkUGyUvvc71XW7jEOogiDgn1ZMoj9Y58LSweUB8vfkY3VF84Fy
zGVvLFWnFbtnWuoQC4OOXv1F6ETdmEsSMEMWKzRLDGPyqNX7FhG3cejlEuiK
EsT5oQhz00RECmnR8mJkGmJhzWmmeoQvaucFqhTOZjvYl9ivuEMBZ2jtkdHn
R6UNcNZRpCbfnl7YoikqbGBgqDFYydFGHXKHDpYhcQYWJsMDUayzUiPtzmaE
tgfTgNNqgPxkLnWA99KYNU8DH+FwgaYBuw9dwdqwcjxSbZjHhCFGUfsqM+ik
O9gcoAwVO1usODlESU7OWuSF2tIv5DBG5rhlSyxBp4d4aWmaumTswAdojy2O
sM70ETbg0mC8niW9lNsgJp55oFmlksvzjUIit5rBEr55rtPcJSgakPpR2yvd
Q4XjybUmV0IdONMhk/OlqaBPtnA0RG+qTcm7oP1qH/m5zRA0ZplVQ5ylkQwr
LwGq5JQpJkgxxgLIrUbbtzYBShXSr5c1XXR0LIRiNgtb6s1s4mt+fbyExdJF
+ceuu+/xYrW/YTuEJpHxLiZ+aNPW5g5Y7Hbqu3hp3UL/kD44cc8JjZh18spX
p/ncojDhF2r2vhR+CndYcMkpGMV+t1pGKC8wlcFc7lb9GJASnqMGvhQmvIaG
gO/x7UjuTZSBW+kITHHZJDqYryKUv1j0CkHxIn9tWsYOpa1giOFtXX5v0AAM
AJR/s/beLlqwCsUdYnw1TkP/0u0ZK3RPio1nJ7S6ckPfsM7DqCWD8ILD8Cdr
cuzQrWaE30t5PXx53xBPO+6t5wKfDL35WHWG1Irmvz9UuT7tDS3IzwtF4ijF
PX6alTbxGnoHgZ4bG4J1wfpTNPppP1gJeVg67VqOypzdZi+SjofMWnFgRFmD
yEN8xpFUs7A9xryVZOosp9Sfe3IbBkO99sAQ7jV4EoMYk3/GKA==
=LjkG
-----END PGP MESSAGE-----",
]
`;
